src/execCmd.ts

```diff
@@ -0,0 +1,210 @@
+ */
+
+import { join } from 'path';
+import { inspect } from 'util';
+import { fs as fsCore } from '@salesforce/core';
+import { Duration, env, parseJson } from '@salesforce/kit';
+import { AnyJson, isNumber } from '@salesforce/ts-types';
+import Debug from 'debug';
+import * as shelljs from 'shelljs';
+import { ExecCallback, ExecOptions, ShellString } from 'shelljs';
+
+export interface ExecCmdOptions extends ExecOptions {
+  /**
+   * Throws if this exit code is not returned by the child process.
+   */
+  ensureExitCode?: number;
+}
+
+export interface ExecCmdResult {
+  /**
+   * Command output from the shell.
+   *
+   * @see https://www.npmjs.com/package/shelljs#execcommand--options--callback
+   */
+  shellOutput: ShellString;
+
+  /**
+   * Command output parsed as JSON, if `--json` param present.
+   */
+  jsonOutput?: AnyJson;
+
+  /**
+   * The JsonParseError if parsing failed.
+   */
+  jsonError?: Error;
+
+  /**
+   * Command execution duration.
+   */
+  execCmdDuration: Duration;
+}
+
+const DEFAULT_SHELL_OPTIONS = {
+  timeout: 300000, // 5 minutes
+  cwd: process.cwd(),
+  env: Object.assign({}, process.env),
+  silent: true,
+};
+
+// Create a Duration instance from process.hrtime
+const hrtimeToMillisDuration = (hrTime: [number, number]) =>
+  Duration.milliseconds(hrTime[0] * Duration.MILLIS_IN_SECONDS + hrTime[1] / 1e6);
+
+// Add JSON output if json flag is set
+const addJsonOutput = (cmd: string, result: ExecCmdResult): ExecCmdResult => {
+  if (cmd.includes('--json')) {
+    try {
+      result.jsonOutput = parseJson(result.shellOutput.stdout);
+    } catch (parseErr: unknown) {
+      result.jsonError = parseErr as Error;
+    }
+  }
+  return result;
+};
+
+const getExitCodeError = (expectedCode: number, actualCode: number, cmd: string) => {
+  return Error(`Unexpected exit code for command: ${cmd}. Expected: ${expectedCode} Actual: ${actualCode}`);
+};
+
+/**
+ * Synchronously execute a command with the provided options in a child process.
+ *
+ * Option defaults:
+ *    1. `cwd` = process.cwd()
+ *    2. `timeout` = 300000 (5 minutes)
+ *    3. `env` = process.env
+ *    4. `silent` = true (child process output not written to the console)
+ *
+ * Other defaults:
+ *
+ *    @see www.npmjs.com/package/shelljs#execcommand--options--callback
+ *    @see www.nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback
+ *
+ * @param cmd The command string to be executed by a child process.
+ * @param options The options used to run the command.
+ * @returns The child process exit code, stdout, stderr, cmd run time, and the parsed JSON if `--json` param present.
+ */
+export const execCmd = (cmd: string, options: ExecCmdOptions = {}): ExecCmdResult => {
+  const debug = Debug('testkit:execCmd');
+
+  // Ensure we run synchronously
+  if (options.async) {
+    throw new Error('execCmd must be run synchronously.  Use execCmdAsync to run asynchronously.');
```

Comment:
Oops, I got it backwards.  Yeah I can change it to `execCmdSync`.

Not sure about the sig overriding.  For some reason I didn't think it would work when I first starting writing it but maybe I'm wrong.  I might have to play around with `ExecCmdOptions` type a bit.  I kinda like making it explicit anyway.